hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieWriteHandle.java

```diff
@@ -81,18 +86,24 @@ public abstract class HoodieWriteHandle<T, I, K, O> extends HoodieIOHandle<T, I,
   protected final TaskContextSupplier taskContextSupplier;
   // For full schema evolution
   protected final boolean schemaOnReadEnabled;
+  protected final boolean preserveMetadata;
+  /**
+   * Flag saying whether secondary index streaming writes is enabled for the table.
+   */
+  protected final boolean isSecondaryIndexStatsStreamingWritesEnabled;
+  List<Pair<String, HoodieIndexDefinition>> secondaryIndexDefns = Collections.emptyList();
```

Comment:
The List is now `final` and always empty, this seems like a bug

Reply:
it is initialized in `#initMetadataPartitionsToCollectStats`, for table service it is empty because the flag `preserveMetadata` is set up as `true`.